ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -127,6 +135,13 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
 
   const dispatch = useDispatch();
 
+  useMount(() => {
+    if (state) {
+      // Restoring state from state;
+      state.expanded.forEach((path) => fetchingPathsRef.current.push(path));
+    }
+  });
+
```

Comment:
We don't need an effect for this.
```suggestion
if (state && fetchingPathsRef.current === null) {
  // Restoring previously loaded state from redux
  state.expanded.forEach((path) => fetchingPathsRef.current.push(path));
} else {
  fetchingPathsRef.current = [];
}
```
Just make sure to initialize the ref in null for this to work.